fili-core/src/test/groovy/com/yahoo/bard/webservice/table/PermissiveConcretePhysicalTableSpec.groovy

```diff
@@ -0,0 +1,100 @@
+// Copyright 2017 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.table
+
+import static com.yahoo.bard.webservice.data.time.DefaultTimeGrain.DAY
+import static org.joda.time.DateTimeZone.UTC
+
+import com.yahoo.bard.webservice.data.config.names.TableName
+import com.yahoo.bard.webservice.data.dimension.BardDimensionField
+import com.yahoo.bard.webservice.data.dimension.Dimension
+import com.yahoo.bard.webservice.data.dimension.DimensionColumn
+import com.yahoo.bard.webservice.data.dimension.MapStoreManager
+import com.yahoo.bard.webservice.data.dimension.impl.KeyValueStoreDimension
+import com.yahoo.bard.webservice.data.dimension.impl.ScanSearchProviderManager
+import com.yahoo.bard.webservice.data.metric.MetricColumn
+import com.yahoo.bard.webservice.metadata.TestDataSourceMetadataService
+import com.yahoo.bard.webservice.table.resolver.DataSourceConstraint
+import com.yahoo.bard.webservice.util.SimplifiedIntervalList
+
+import org.joda.time.Interval
+
+import spock.lang.Shared
+import spock.lang.Specification
+import spock.lang.Unroll
+
+class PermissiveConcretePhysicalTableSpec extends Specification {
+    @Shared PermissiveConcretePhysicalTable permissiveConcretePhysicalTable
+
+    @Shared TableName tableName
+
+    @Shared Set<Column> columns
+
+    @Shared Dimension dimension
+    @Shared DimensionColumn dimensionColumn
+    @Shared MetricColumn metricColumn1, metricColumn2
+
+    @Shared Interval interval1, interval2, interval3
+
+    def setupSpec() {
```

Comment:
Ahh, right, it's because `@Shared` is needed to be able to reference variables in the `where` block. I consider this a Spock bug :(